### PR TITLE
[SD-234] Crossed out compare at price

### PR DESCRIPTION
**Pull request by aplegatt**

### CODE REVIEW

File: frontend/app/views/spree/products/_cart_form.html.erb [hunk 1]
```diff
@@ -27,6 +27,11 @@
   <div id="inside-product-cart-form" data-hook="inside_product_cart_form">
     <% if is_product_available_in_currency %>
       <div id="product-price" class="mb-2 text-center text-md-left add-to-cart-form-price" data-hook="product_price">
+        <span class="pre-sales-price">
+          <% if !default_variant&.pre_sales_price.nil? && !(default_variant&.pre_sales_price <= default_variant.price) %>
```
Comment (damianlegawiec) [inline frontend/app/views/spree/products/_cart_form.html.erb:31]:
```suggestion
          <% if default_variant.pre_sales_price.present? && default_variant.pre_sales_price < default_variant.price %>
```

there's always a `default_variant`

File: backend/app/views/spree/admin/variants/_form.html.erb [hunk 2]
```diff
@@ -24,6 +24,13 @@
         <%= f.text_field :price, value: number_to_currency(@variant.price, unit: ''), class: 'form-control' %>
       </div>
 
+      <% if type == 'edit' %>
```
Comment (damianlegawiec) [inline backend/app/views/spree/admin/variants/_form.html.erb:27]: @aplegatt it also needs to be visible when you add a new variant

File: core/app/models/spree/variant.rb [hunk 3]
```diff
@@ -217,6 +217,15 @@ def price_modifier_amount(options = {})
       end.sum
     end
 
+    def compare_at_price
```
Comment (damianlegawiec) [inline core/app/models/spree/variant.rb:220]: @aplegatt please implement this in https://github.com/spree/spree/blob/d2a72bca801d223eec6eb899d146f2586ba44412/core/app/models/concerns/spree/default_price.rb#L11

File: core/app/presenters/spree/variant_presenter.rb [hunk 4]
```diff
@@ -71,5 +73,13 @@ def variant_attributes(variant)
         purchasable: variant.purchasable?
       }
     end
+
+    def should_display_compare_at_price(variant)
+      if variant.compare_at_price.nil?
+        false
+      else
+        variant.compare_at_price > variant.price
+      end
```
Comment (damianlegawiec) [inline core/app/presenters/spree/variant_presenter.rb:82]:
```suggestion
      variant.compare_at_price.present? && variant.compare_at_price > variant.price
```

File: frontend/app/assets/javascripts/spree/frontend/views/spree/products/cart_form.js [hunk 5]
```diff
@@ -220,7 +221,12 @@ function CartForm($, $cartForm) {
 
     if (!variant) return
 
+    var shouldDisplayCompareAtPrice = variant.should_display_compare_at_price
+
     this.$price.html(variant.display_price)
+
+    var compareAtPriceContent = shouldDisplayCompareAtPrice ? `<span class="mr-3">${variant.display_compare_at_price}</span>` : ''
```
Comment (AleksandraKaminska) [inline frontend/app/assets/javascripts/spree/frontend/views/spree/products/cart_form.js:228]: Could you use es5 string interpolation instead?

**Review by AleksandraKaminska — APPROVED** [review]

**Pull request merged**